velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/substrait/VeloxToSubstraitExpr.h"
+
+#include "velox/expression/ControlExpr.h"
+
+namespace facebook::velox::substrait {
+
+const ::substrait::Expression& VeloxToSubstraitExprConvertor::toSubstraitExpr(
+    google::protobuf::Arena& arena,
+    const std::shared_ptr<const ITypedExpr>& expr,
+    const RowTypePtr& inputType) {
+  ::substrait::Expression* substraitExpr =
+      google::protobuf::Arena::CreateMessage<::substrait::Expression>(&arena);
+  if (auto constExpr =
+          std::dynamic_pointer_cast<const ConstantTypedExpr>(expr)) {
+    substraitExpr->mutable_literal()->MergeFrom(
+        toSubstraitExpr(arena, constExpr));
+    return *substraitExpr;
+  }
+  if (auto callTypeExpr =
+          std::dynamic_pointer_cast<const CallTypedExpr>(expr)) {
```

Comment:
Let's make sure tests cover all these expressions, e.g. call, constant and cast.

Reply:
Yes, the test about filter convertor (`auto plan = PlanBuilder().values(vectors).filter("c2 < 1000").planNode();`) has covered all these expressions.